sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala

```diff
@@ -56,6 +58,7 @@ abstract class FileFormatDataWriter(
     if (currentWriter != null) {
```

Comment:
I'd like to add a new method in the base class
```
protected def releaseCurrentWriter() ...
```
Then `releaseResources` simply calls `releaseCurrentWriter`. The child should call `releaseCurrentWriter` when needed, instead of `super[FileFormatDataWriter].releaseResources()`, which looks pretty weird.

Reply:
@cloud-fan - yes I agree with that looks pretty weird. Good idea and updated.